### PR TITLE
Fix #11650 add an error message for no source files parsing a tsconfig.json

### LINKED ISSUE #11650: Error for compilataions with no source files
Running tsc on an empty input set should result into an error. Most of the time this is an error in the configurations that the user did not intend. 

For instance, see https://github.com/Microsoft/TypeScript/issues/11621. 

```json
  {
      "compilerOptions": {
          "outDir": "src"
      },
      "include":[
          "src/**/*"
      ]
  }
```

This config file results in **no** input files, yet calling `tsc` on it produces no errors.

### PR DESCRIPTION
**Pull request by mhegazy**
Fixes #11650

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -3085,6 +3086,15 @@
         "category": "Error",
         "code": 18001
     },
+    "No input files were found in config file: '{0}'. 'files' list is empty.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: `The 'files' list in config file '{0}' is empty.`

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -3085,6 +3086,15 @@
         "category": "Error",
         "code": 18001
     },
+    "No input files were found in config file: '{0}'. 'files' list is empty.": {
+        "category": "Error",
+        "code": 18002
+    },
+    "No input files were found in config file: '{0}'. Here are the contents of include: {1} and exclude: {2} lists.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: `No inputs were found in config file '{0}'. Specified 'include' paths were '{1}' and 'exclude' paths were '{2}'.`

Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: they are not really "specified" some of them are inferred.

**Comment by mhegazy** [comment]
@DanielRosenwasser any comments?

**Pull request merged**